presto-main/src/test/java/com/facebook/presto/security/TestFileBasedSystemAccessControl.java

```diff
@@ -37,16 +45,70 @@
 
 public class TestFileBasedSystemAccessControl
 {
-    private static final Principal PRINCIPAL = new TestingPrincipal("principal");
-    private static final Identity alice = new Identity("alice", Optional.of(PRINCIPAL));
-    private static final Identity bob = new Identity("bob", Optional.of(PRINCIPAL));
-    private static final Identity admin = new Identity("admin", Optional.of(PRINCIPAL));
-    private static final Identity nonAsciiUser = new Identity("\u0194\u0194\u0194", Optional.of(PRINCIPAL));
+    private static final Principal TESTING_PRINCIPAL = new TestingPrincipal("principal");
+    private static final Identity alice = new Identity("alice", Optional.of(TESTING_PRINCIPAL));
+    private static final Identity bob = new Identity("bob", Optional.of(TESTING_PRINCIPAL));
+    private static final Identity admin = new Identity("admin", Optional.of(TESTING_PRINCIPAL));
+    private static final Identity nonAsciiUser = new Identity("\u0194\u0194\u0194", Optional.of(TESTING_PRINCIPAL));
     private static final Set<String> allCatalogs = ImmutableSet.of("secret", "open-to-all", "all-allowed", "alice-catalog", "allowed-absent", "\u0200\u0200\u0200");
     private static final QualifiedObjectName aliceTable = new QualifiedObjectName("alice-catalog", "schema", "table");
     private static final QualifiedObjectName aliceView = new QualifiedObjectName("alice-catalog", "schema", "view");
     private static final CatalogSchemaName aliceSchema = new CatalogSchemaName("alice-catalog", "schema");
+    public static final String DEFAULT_CONFIG = "catalog.json";
     private TransactionManager transactionManager;
+    private Principal ldapPrincipal;
+    private Principal kerberosHostRealm;
+    private Principal kerberosIpRealm;
+    private Principal kerberosRealm;
+    private Principal kerberosHost;
+    private Principal kerberosSimple;
+    private Properties oldSessionProperties;
+
+    @BeforeClass
+    public void setUp()
+    {
+        oldSessionProperties = setKerberosSessionProperties();
+        try {
+            ldapPrincipal = new LdapPrincipal("CN=principal");
+        }
+        catch (InvalidNameException e) {
+            fail(e.toString());
```

Comment:
Why you need that? Is it for better readability of the error message?
Because whenever `LdapPrincipal` c-tor throws, `setUp` will be interrupted regardless of this `catch + fail` block, so the test will fail.